String.cpp

```diff
@@ -0,0 +1,204 @@
+//Copyright 2020 XXXPUSSYDESTROYERXXX
+#include "string.hpp"
+
+String::~String() {
+  delete[] Data;
+}
+
+String::String() {
+  Data = new char[10];
+  capacity = 10;
+  length = 0;
+}
+
+String::String(const String& rhs) {
+  Data = new char[rhs.capacity];
+  length = rhs.length;
+  capacity = rhs.capacity;
+
+  for (size_t i = 0; i < length; i++){
+    Data[i] = rhs.Data[i];
+  }
+}
+
+String::String(const char* data) {
+  //finding length
+  length = 0;
+  while (data[length] != '\0') length++;
+
+  Data = new char[length + 10];
+  capacity = length + 10;
+
+  for (size_t i = 0; i < length; i++)
+    Data[i] = data[i];
+}
+
+String& String::operator=(const String& rhs){
```

Comment:
See cppreference for good implementation example.
`rhs` may be the same as this object thus making this method broken